tests/tasks/cubejs/test_cubejs_tasks.py

```diff
@@ -0,0 +1,135 @@
+
+    def test_run_without_api_secret_api_secret_env_var(self):
+        cubejs_task = CubeJSQueryTask()
+        with pytest.raises(ValueError) as exc:
+            cubejs_task.run(subdomain="foo")
+
+        assert "Missing `api_secret` and `api_secret_env_var` not found." in str(exc)
+
+    def test_run_without_query_raises(self):
+        cubejs_task = CubeJSQueryTask()
+        with pytest.raises(ValueError) as exc:
+            cubejs_task.run(subdomain="foo", api_secret="bar")
+
+        assert "Missing `query`." in str(exc)
+
+    @responses.activate
+    def test_run_with_failing_api_raises(self):
+        cubejs_task = CubeJSQueryTask()
+        responses.add(
+            responses.GET, "https://test.cubecloud.dev/cubejs-api/v1/load", status=123
+        )
+
+        with pytest.raises(FAIL) as exc:
+            cubejs_task.run(subdomain="test", api_secret="foo", query="query")
+
+        assert "Cube.js load API failed!." in str(exc)
+
+    @responses.activate
+    def test_run_with_continue_waiting(self, caplog):
+        caplog.set_level(logging.DEBUG)
+        cubejs_task = CubeJSQueryTask()
+
+        responses.add(
+            responses.GET,
+            "https://test.cubecloud.dev/cubejs-api/v1/load",
+            status=200,
+            json={"error": "Continue wait"},
+        )
+
+        responses.add(
+            responses.GET,
+            "https://test.cubecloud.dev/cubejs-api/v1/load",
+            status=200,
+            json={"data": "result"},
+        )
+
+        data = cubejs_task.run(subdomain="test", api_secret="foo", query="query")
+
+        assert "Cube.js load API still running." in caplog.text
+        assert isinstance(data, dict)
+
+    @responses.activate
+    def test_run_with_security_context(self, caplog):
+        caplog.set_level(logging.DEBUG)
+        cubejs_task = CubeJSQueryTask()
+
+        responses.add(
+            responses.GET,
+            "https://test.cubecloud.dev/cubejs-api/v1/load",
+            status=200,
+            json={"data": "result"},
+        )
+
+        cubejs_task.run(
+            subdomain="test",
+            api_secret="foo",
+            query="query",
+            security_context={"foo": "bar"},
+        )
+
+        assert "JWT token generated with security context." in caplog.text
+
+    @responses.activate
+    def test_run_with_max_wait_time_raises(self):
+        cubejs_task = CubeJSQueryTask()
+
+        responses.add(
+            responses.GET,
+            "https://test.cubecloud.dev/cubejs-api/v1/load",
+            status=200,
+            json={"error": "Continue wait"},
+        )
+
+        with pytest.raises(FAIL) as exc:
+            cubejs_task.run(
+                subdomain="test",
+                api_secret="foo",
+                query="query",
+                security_context={"foo": "bar"},
+                max_wait_time=15,
+            )
+
+        assert "Cube.js load API took too long to provide a response." in str(exc)
```

Comment:
Having a configurable time between calls would be useful for this test since it could spend less time waiting.